market/mysterium/mysterium_api.go

```diff
@@ -151,43 +151,37 @@ func (mApi *MysteriumAPI) PingProposal(proposal market.ServiceProposal, signer i
 	return err
 }
 
-// GetProposal fetches service proposal from discovery by exact ID
-func (mApi *MysteriumAPI) GetProposal(id market.ProposalID) (*market.ServiceProposal, error) {
-	proposals, err := mApi.doProposalRequest(url.Values{
-		"node_key":     {id.ProviderID},
-		"service_type": {id.ServiceType},
+// Proposals fetches currently active service proposals from discovery
+func (mApi *MysteriumAPI) Proposals() ([]market.ServiceProposal, error) {
+	return mApi.QueryProposals(ProposalsQuery{
+		ServiceType:     "all",
+		AccessPolicyAll: true,
 	})
-	if err != nil {
-		return nil, err
-	}
-	if len(proposals) == 0 {
-		return nil, nil
-	}
-
-	return &proposals[0], nil
 }
 
-// FindProposals fetches currently active service proposals from discovery
-func (mApi *MysteriumAPI) FindProposals(filter market.ProposalFilter) ([]market.ServiceProposal, error) {
-	query := url.Values{}
-	if filter.ProviderID != "" {
-		query.Set("node_key", filter.ProviderID)
+// QueryProposals fetches currently active service proposals from discovery - by given query filter
+func (mApi *MysteriumAPI) QueryProposals(query ProposalsQuery) ([]market.ServiceProposal, error) {
+	values := url.Values{}
+	if query.NodeKey != "" {
+		values.Set("node_key", query.NodeKey)
 	}
-	if filter.ServiceType != "" {
-		query.Set("service_type", filter.ServiceType)
+	if query.ServiceType != "" {
+		values.Set("service_type", query.ServiceType)
 	}
-	if filter.AccessPolicyID != "" {
-		query.Set("access_policy[id]", filter.AccessPolicyID)
+	if query.AccessPolicyAll {
+		values.Set("access_policy", "*")
```

Comment:
We have 
`"all"` for `ServiceType`
`"*"` for `AccessPolicy`
`bool` as `AccessPolicyAll`

This is very confusing.

Reply:
Yes it is, that's why I do this PR, so that we isolate this behaviour and don't call `MysteriumAPI.FindProposals()` anumore